app/src/main/java/com/example/userinfo/MainActivity.kt

```diff
@@ -0,0 +1,148 @@
+        ) {
+            Toast.makeText(this, "Every field is mandatory", Toast.LENGTH_LONG).show()
+            return false
+
+        } else {
+            return true
+        }
+    }
+
+    private fun phoneNumberValidation(): Boolean {
+        if (binding.phoneNumberET.text.toString().length == 10) {
+            return true
+        } else {
+            Toast.makeText(this, "Phone number ( should be only length of 10) ", Toast.LENGTH_LONG)
+                .show()
+            return false
+        }
+    }
+
+    private fun pinCodeValidation(): Boolean {
+        if (binding.pinCodeET.text.toString().length == 6) {
+            return true
+        } else {
+            Toast.makeText(this, "Pin code ( should be only length of 6) ", Toast.LENGTH_LONG)
+                .show()
+            return false
+        }
+    }
+
+    private fun emailValidation(): Boolean {
+        if ((binding.emailET.text.toString().contains("@")) && ((binding.emailET.text.toString()
+                .endsWith(".com")) || (binding.emailET.text.toString().endsWith("co.in")))
+        ) {
+            return true
+        } else {
+            Toast.makeText(this, " Email should have @ and .com or co.in", Toast.LENGTH_LONG).show()
+            return false
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(USERNAME, binding.validUserName.text.toString())
+        outState.putString(EMAIL, binding.validEmail.text.toString())
+        outState.putString(PHONENUMBER, binding.validPhoneNumber.text.toString())
+        outState.putString(PINCODE, binding.validPinCode.text.toString())
+        outState.putString(ADDRESS, binding.validAddress.text.toString())
+        outState.putBoolean(VISIBLE, visibleFieldsCheck)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        if (savedInstanceState.getBoolean(VISIBLE)) {
+            binding.validUserName.setText(savedInstanceState.getString(USERNAME))
+            binding.validEmail.setText(savedInstanceState.getString(EMAIL))
+            binding.validPhoneNumber.setText(savedInstanceState.getString(PHONENUMBER))
+            binding.validPinCode.setText(savedInstanceState.getString(PINCODE))
+            binding.validAddress.setText(savedInstanceState.getString(ADDRESS))
+            fieldsDisplayOnValidate(savedInstanceState.getBoolean(VISIBLE))
+            visibleFieldsCheck = true
+            binding.confirm.setOnClickListener {
+                dataPassingFromActivityToActivityOnConfirm()
+            }
+            binding.cancel.setOnClickListener {
+                fieldsDisplayOnValidate(false)
+                visibleFieldsCheck = false
+            }
+
+        }
+        super.onRestoreInstanceState(savedInstanceState)
+
+    }
+
+    private fun fieldsDisplayOnValidate(visible: Boolean) {
+        binding.validUserName.isVisible = visible
```

Comment:
We can make use of viewgroups visibility here
For example
All validViewsLayout.visiblity = visible
where `validViewsLayout (LinearLayout or Group if you are usign ConstraintLayout- https://developer.android.com/reference/androidx/constraintlayout/widget/Group)` contains validUserName, validEmail, validPhoneNumber, validPinCode and validAddress